src/raftstore/store/snap.rs

```diff
@@ -765,41 +792,35 @@ impl SnapManagerCore {
             }
         }
 
-        self.notify_stats();
+        notify_stats(self.ch.as_ref());
     }
 
-    pub fn deregister(&mut self, key: &SnapKey, entry: &SnapEntry) {
+    pub fn deregister(&self, key: &SnapKey, entry: &SnapEntry) {
         debug!("deregister [key: {}, entry: {:?}]", key, entry);
         let mut need_clean = false;
         let mut handled = false;
-        if let Some(e) = self.registry.get_mut(key) {
+        let mut core = self.core.wl();
+        if let Some(e) = core.registry.get_mut(key) {
             let last_len = e.len();
             e.retain(|e| e != entry);
             need_clean = e.is_empty();
             handled = last_len > e.len();
         }
         if need_clean {
-            self.registry.remove(key);
+            core.registry.remove(key);
         }
         if handled {
-            self.notify_stats();
+            notify_stats(self.ch.as_ref());
             return;
         }
         warn!("stale deregister key: {} {:?}", key, entry);
     }
 
-    fn notify_stats(&self) {
```

Comment:
Because this function only need to borrow `self.ch`, but if it's implemented as a method, it will borrow the whole `self`, so there will be an extra scope at line 781 and line 802, which makes code ugly.